packages/preferences/src/browser/views/components/preference-number-input.tsx

```diff
@@ -59,18 +58,19 @@ export const PreferenceNumberInput: React.FC<PreferenceNumberInputProps> = ({ pr
      * Validates the input.
      * @param input the input value.
      */
-    const getInputValidation = (input: number | undefined): { isValid: boolean, message: string } => {
+    const getInputValidation = (input: string): { isValid: boolean, message: string } => {
         const errorMessages: string[] = [];
-        if (!input) {
+        if (input === '') {
```

Comment:
Why not keep the `input` parameter as `number` and instead check for `isNaN` here? If the call to `Number(newValue)` didn't convert into a number it would be `NaN`.

Reply:
Ah, an empty field would be evaluated to `0` using the return from `Number(newValue)`...

Reply:
> Ah, an empty field would be evaluated to 0 using the return from Number(newValue)...

Right, I need to make the distinction between `0` and an empty field `''`.

Due to the following, I need to deal with `strings` directly from the input first, then use the number conversion when doing arithmetic operations:

```bash
$ node
> Number('')
0
> Number(0)
0
> isNaN('')
false
```